openproblems/tasks/spatial_decomposition/methods/nmfreg.py

```diff
@@ -0,0 +1,97 @@
+from ....tools.decorators import method
+from ....tools.normalize import log_cpm
+from ....tools.utils import check_version
+from sklearn.decomposition import NMF
+from sklearn.preprocessing import StandardScaler
+
+import collections
+import numpy as np
+import pandas as pd
+import scipy.optimize
+import scipy.stats
+
+
+@method(
+    method_name="NMF-reg",
+    paper_name="Slide-seq: A scalable technology for measuring genome-wide expression at high spatial resolution",
+    paper_url="https://science.sciencemag.org/content/363/6434/1463",
+    paper_year=2009,
+    code_url="N/A",
+    code_version=check_version("NMF-reg"),
+)
+def nmfreg(adata):
+    n_types = adata.obsm["proportions_true"].size[1]
+    adata_sc = adata.uns["sc_reference"].copy()
+    factors = 30
+    projectiob_type = "l2"
+
+    # Learn from reference
+    X = adata_sc.X.toarray()
+    X_norm = X / X.sum(1)[:, np.newaxis]
```

Comment:
if you are trying to normalize raw counts, there are utils methods for that (so that it's consisnte across tasks). For instance, in this case I think this is the recommended function:
```python
from ....tools.normalize import 
...
log_cpm(adata)
```

there are more in `tools.normalize` that could be useful (and prob important to test?)

Reply:
ok I saw the methods available, does not seem to do what we are trying to do here. Should I create a function in tools.normalize file @giovp? I guess this one is very method-specific.

Reply:
I see ok, I think I'd just keep it as is then as indeed is method specific.